Shell/TermAlgebra.cpp

```diff
@@ -26,6 +26,7 @@ TermAlgebraConstructor::TermAlgebraConstructor(unsigned functor, Lib::Array<unsi
   : _functor(functor), _hasDiscriminator(false), _destructors(destructors)
 {
   _type = env.signature->getFunction(_functor)->fnType();
+#if VDEBUG
```

Comment:
Remaining stuff in the function is debug-only checks. Perhaps we should also change the signature of the constructor to not require this stuff in release, but it's a minor optimisation.